search-commons/src/main/java/no/unit/nva/search2/common/OpenSearchSwsClient.java

```diff
@@ -18,50 +22,76 @@
 import java.net.http.HttpResponse;
 import java.nio.charset.StandardCharsets;
 
+import static no.unit.nva.auth.AuthorizedBackendClient.AUTHORIZATION_HEADER;
 import static no.unit.nva.auth.uriretriever.AuthorizedBackendUriRetriever.ACCEPT;
 import static no.unit.nva.search.RestHighLevelClientWrapper.SEARCH_INFRASTRUCTURE_CREDENTIALS;
 import static no.unit.nva.search2.constant.ApplicationConstants.objectMapperWithEmpty;
 import static no.unit.nva.search2.constant.ApplicationConstants.readSearchInfrastructureAuthUri;
 import static nva.commons.core.attempt.Try.attempt;
-import static org.apache.http.entity.ContentType.APPLICATION_JSON;
 
 public class OpenSearchSwsClient {
 
     private static final Logger logger = LoggerFactory.getLogger(OpenSearchSwsClient.class);
     private static final String REQUESTING_SEARCH_FROM = "OpenSearchSwsClient url -> {}";
-    private final AuthorizedBackendClient contentRetriever;
-    private final String mediaType;
+    private final CachedJwtProvider jwtProvider;
+    private final HttpClient httpClient;
+    private final BodyHandler<String> bodyHandler;
 
-    public OpenSearchSwsClient(AuthorizedBackendClient contentRetriever, String mediaType) {
-        this.contentRetriever = contentRetriever;
-        this.mediaType = mediaType;
+    public OpenSearchSwsClient(CachedJwtProvider jwtProvider, HttpClient client) {
+        this.bodyHandler = HttpResponse.BodyHandlers.ofString(StandardCharsets.UTF_8);
+        this.jwtProvider = jwtProvider;
+        this.httpClient = client;
     }
 
     @JacocoGenerated
     public static OpenSearchSwsClient defaultSwsClient() {
-        var uri = URI.create(readSearchInfrastructureAuthUri());
-        var credentials
-            = new SecretsReader()
-            .fetchClassSecret(SEARCH_INFRASTRUCTURE_CREDENTIALS, UsernamePasswordWrapper.class);
-
-        var cognitoCredentials = new CognitoCredentials(credentials::getUsername, credentials::getPassword, uri);
-        var cognitoAuthenticator = CognitoAuthenticator.prepareWithCognitoCredentials(cognitoCredentials);
-        var cachedJwtProvider = CachedJwtProvider.prepareWithAuthenticator(cognitoAuthenticator);
-        var retriever = AuthorizedBackendClient.prepareWithBearerToken(cachedJwtProvider.getValue().getToken());
```

Comment:
This seems to have been the bug. You only ever called cachedJwtProvider.getValue() once; upon initialization. I see you now instead do it on each request - good work!